crates/bevy_core_pipeline/src/smaa/smaa.wgsl

```diff
@@ -0,0 +1,1122 @@
+    factor.x -= rounding.y * sample_edges_level_zero_offset(tex_coord.zw, vec2( 1, 1)).g;
+    factor.y -= rounding.x * sample_edges_level_zero_offset(tex_coord.xy, vec2(-2, 0)).g;
+    factor.y -= rounding.y * sample_edges_level_zero_offset(tex_coord.zw, vec2(-2, 1)).g;
+
+    return weights * saturate(factor);
+#else   // SMAA_DISABLE_CORNER_DETECTION
+    return weights;
+#endif  // SMAA_DISABLE_CORNER_DETECTION
+}
+
+//-----------------------------------------------------------------------------
+// Blending Weight Calculation Pixel Shader (Second Pass)
+
+@fragment
+fn blending_weight_calculation_fragment_main(in: BlendingWeightCalculationVaryings)
+        -> @location(0) vec4<f32> {
+    let subsample_indices = vec4(0.0);  // Just pass zero for SMAA 1x, see @SUBSAMPLE_INDICES.
+
+    var weights = vec4(0.0);
+
+    var e = textureSample(edges_texture, edges_sampler, in.tex_coord).rg;
+
+    if (e.g > 0.0) {    // Edge at north
+#ifndef SMAA_DISABLE_DIAG_DETECTION
+        // Diagonals have both north and west edges, so searching for them in
+        // one of the boundaries is enough.
+        weights = vec4(calculate_diag_weights(in.tex_coord, e, subsample_indices), weights.ba);
+
+        // We give priority to diagonals, so if we find a diagonal we skip 
+        // horizontal/vertical processing.
+        let diagonal_found = weights.r + weights.g != 0.0;
+#else   // SMAA_DISABLE_DIAG_DETECTION
+        let diagonal_found = false;
+#endif  // SMAA_DISABLE_DIAG_DETECTION
+
+        if (!diagonal_found) {
+            var d: vec2<f32>;
+
+            // Find the distance to the left:
+            var coords: vec3<f32>;
+            coords.x = search_x_left(in.offset_0.xy, in.offset_2.x);
+            // in.offset_1.y = in.tex_coord.y - 0.25 * smaa_info.rt_metrics.y (@CROSSING_OFFSET)
+            coords.y = in.offset_1.y;
+            d.x = coords.x;
+
+            // Now fetch the left crossing edges, two at a time using bilinear
+            // filtering. Sampling at -0.25 (see @CROSSING_OFFSET) enables to
+            // discern what value each edge has:
+            let e1 = textureSampleLevel(edges_texture, edges_sampler, coords.xy, 0.0).r;
+
+            // Find the distance to the right:
+            coords.z = search_x_right(in.offset_0.zw, in.offset_2.y);
+            d.y = coords.z;
+
+            // We want the distances to be in pixel units (doing this here allow to
+            // better interleave arithmetic and memory accesses):
+            d = abs(round(smaa_info.rt_metrics.zz * d - in.position.xx));
+
+            // SMAAArea below needs a sqrt, as the areas texture is compressed
+            // quadratically:
+            let sqrt_d = sqrt(d);
+
+            // Fetch the right crossing edges:
+            let e2 = textureSampleLevel(
+                edges_texture, edges_sampler, texture_offset(coords.zy, vec2<i32>(1, 0)), 0.0).r;
```

Comment:
Unnecessary, wgsl has an offset param for [`textureSampleLevel`](https://gpuweb.github.io/gpuweb/wgsl/#texturesamplelevel)

```suggestion
            let e2 = textureSampleLevel(edges_texture, edges_sampler, coords.zy, 0.0, vec2<i32>(1, 0)).r;
```